src/main/webapp/app/course/learning-paths/components/learning-path-lecture-unit/learning-path-lecture-unit.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Signal, computed, inject, input, output } from '@angular/core';
+import { toObservable, toSignal } from '@angular/core/rxjs-interop';
+import { LectureUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/lectureUnit.service';
+import { Observable, catchError, map, of, switchMap } from 'rxjs';
+import { AlertService } from 'app/core/util/alert.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { onError } from 'app/shared/util/global.utils';
+import { LoadedValue } from 'app/course/learning-paths/components/learning-path-student-nav/learning-path-student-nav.component';
+import { LectureUnit, LectureUnitType } from 'app/entities/lecture-unit/lectureUnit.model';
+import { ArtemisLectureUnitsModule } from 'app/overview/course-lectures/lecture-units.module';
+import { LectureUnitCompletionEvent } from 'app/overview/course-lectures/course-lecture-details.component';
+
+@Component({
+    selector: 'jhi-learning-path-lecture-unit',
+    standalone: true,
+    imports: [ArtemisLectureUnitsModule],
+    templateUrl: './learning-path-lecture-unit.component.html',
+    styleUrl: './learning-path-lecture-unit.component.scss',
+})
+export class LearningPathLectureUnitComponent {
+    private readonly lectureUnitService = inject(LectureUnitService);
+    private readonly alertService = inject(AlertService);
```

Comment:
we normally use constructor injection:
```suggestion
    public constructor(private lectureUnitService: LectureUnitService, private alertService: AlertService) {
    }
```

Reply:
I just used the newly introduced inject() Function to further improve readability